enhancements/proposals/2549-scylladbcluster-backup-repair/README.md

```diff
@@ -0,0 +1,439 @@
+
+The existing ScyllaCluster controller, running as part of the manager-controller, will also be extended to set a `scylla-operator.scylladb.com/scylladb-manager-cluster` finalizer on ScyllaCluster objects so that the clusters are deleted from ScyllaDB Manager state on deletion [#1930][2].
+
+#### Status conditions
+
+The new ScyllaDBCluster controller, running as part of the manager-controller, will wait for ScyllaDBCluster to be rolled out to register it with ScyllaDB Manager.
+
+The existing ScyllaCluster controller, running as part of the manager-controller, will also wait for ScyllaCluster to be rolled out to register it with Scylla Manager.
+
+### Scheduling of backup and repair tasks with ScyllaDB Manager
+
+We introduce a new API, `ScyllaDBManagerTask.scylla.scylladb.com/v1alpha1`, which allows users to schedule backup and repair tasks for ScyllaDBClusters and ScyllaClusters registered with ScyllaDB Manager.
+The existing manager-controller application will be extended with a dedicated controller reconciling ScyllaDBManagerTask objects, responsible for scheduling the specified tasks with ScyllaDB Manager.
+
+The existing controller, reconciling ScyllaClusters, will no longer schedule backup and repair tasks defined in `scyllaclusters.scylla.scylladb.com/v1.spec.backups` and `scyllaclusters.scylla.scylladb.com/v1.spec.repairs`.
+Instead, the existing ScyllaCluster translation controller, running as part of the operator, is extended to convert the backup and repair tasks defined in `scyllaclusters.scylla.scylladb.com/v1.spec.backups` and `scyllaclusters.scylla.scylladb.com/v1.spec.repairs` into the ScyllaDBManagerTask objects and apply them.
+Conversely, the controller will convert ScyllaDBManagerTasks' statuses into ScyllaCluster's status and update it.
+
+#### Definition of `ScyllaDBManagerTask.scylla.scylladb.com/v1alpha1`
+
+```go
+type ScyllaDBManagerTaskType string
+
+const (
+	ScyllaDBManagerTaskTypeBackup ScyllaDBManagerTaskType = "Backup"
+	ScyllaDBManagerTaskTypeRepair ScyllaDBManagerTaskType = "Repair"
+)
+
+type ScyllaDBManagerTaskSchedule struct {
+	// cron specifies the task schedule as a cron expression.
+	// It supports the "standard" cron syntax `MIN HOUR DOM MON DOW`, as used by the Linux utility, as well as a set of non-standard macros: "@yearly", "@annually", "@monthly", "@weekly", "@daily", "@midnight", "@hourly", "@every [+-]?<duration>".
+	// +optional
+	Cron *string `json:"cron,omitempty"`
+
+	// numRetries specifies how many times a scheduled task should be retried before failing.
+	// +optional
+	NumRetries *int64 `json:"numRetries,omitempty"`
+
+	// startDate specifies the start date of the task.
+	// If not set, the task is started immediately.
+	// +optional
+	StartDate *time.Time `json:"startDate,omitempty"`
+}
+
+type ScyllaDBManagerBackupTaskOptions struct {
+	// schedule specifies the schedule on which the backup task is run.
+	ScyllaDBManagerTaskSchedule `json:",inline"`
```

Comment:
nit: maybe it wold make more sense to inline schedule into the `ScyllaDBManagerTaskSpec` instead of all tasks options separately?
It would also correspond closer to SM API, where the task is defined as:
```json
    "Task": {
      "type": "object",
      "properties": {
        ...
        "schedule": {
          "$ref": "#/definitions/Schedule"
        },
        "properties": {
          "type": "object",
          "additionalProperties": true
        }
      }
    },
```

Reply:
It would in terms of aligning with manager API, but it would also force us to handle it conditionally for a task type. It doesn't seem to make much sense with a restore, which is a one-time action, and that we plan to add in the future. Or are there any scenarios where restore on a schedule would make sense?

Reply:
The only schedule option which makes any sense with restore is `--start-date`, but it's still not that useful, so it makes sense to have schedule outside of task spec.